types/chrome/index.d.ts

```diff
@@ -6502,9 +6509,8 @@ declare namespace chrome.runtime {
     export interface InstalledDetails {
         /**
          * The reason that this event is being dispatched.
-         * One of: "install", "update", "chrome_update", or "shared_module_update"
          */
-        reason: string;
+        reason: OnInstalledReason;
```

Comment:
This looks good to me, but any thoughts on doing a simple `"install | "update" | "chrome_update" | "shared_module_update"` type instead? The benefit would be not having to import the enum in order to assign a value to reason.

Reply:
It's probably better to use the enum for maintainability. Also, there's no way someone can access `details.reason.INSTALL`, since TS doesn't recognize the value in the callback as a enum, but as a union type of strings.